File: Day1/day1.js
//READ INPUT
const fs = require('fs');

const lines = fs
    .readFileSync('input.txt', { encoding: 'utf-8'})
    .split('\n')
    .filter((x) => Boolean(x))
    .map((x) => parseInt(x));

//FIRST PROBLEM
const problemPart1 = (data) => {
    let count = 0;
    for (let i = 1; i < data.length - 1; i++) {
        if (data[i] > data[i-1]) {
            count++;
        }
    }
    return count;
}

//SECOND PROBLEM
const problemPart2 = (data) => {
    let count = 0;
    for (let i = 2; i < data.length - 1; i++) {
        if ((data[i-2]+data[i-1]+data[i])<(data[i-1]+data[i]+data[i+1])) {
            count++;
        }
    }
    return count;
}

console.log(problemPart1(lines));
console.log(problemPart2(lines));

File: Day1/day1.py
#INPUT
lines = [int(x) for x in open('input.txt').read().split()]

#FIRST PROBLEM
def problemPart1(data):
    count = 0

    for i in range(1, len(data)):
        if data[i] > data[i-1]:
            count += 1
    print(count)

#SECOND PROBLEM
def problemPart2(data):
    count = 0
    for i in range(2, len(data)-1):
        first = data[i-2] + data[i-1] + data[i]
        second = data[i-1] + data[i] + data[i+1]
        if second > first:
            count += 1
    print(count)

problemPart1(lines)
problemPart2(lines)
